File: hashing/pythonCode/basic.py
hash = {}

n = int(input("Enter the number of elements: "))
arr = list(map(int, input(f"Enter {n} elements: ").split()))

for i in arr:
    if i in hash:
        hash[i] += 1
    else:
        hash[i] = 1

q = int(input("Enter number of queries: "))

for _ in range(q):
    number = int(input("Enter a number to query: "))
    print(hash.get(number, 0))

File: hashing/cppCode/basic.cpp
#include <iostream>
using namespace std;

int hashh[100000000];

int main()
{
    int n;
    cin >> n;
    int arr[n];
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    for(int i = 0; i < n; i++) {
        hashh[arr[i]] += 1;
    }
    
    int q;
    cin >> q;
    while(q--) {
        int number;
        cin >> number;
        cout << hashh[number] << endl;
    }
    return 0;
}
